cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceUtilsTest.java

```diff
@@ -164,4 +167,52 @@ public void testAutoRebalanceStatusNewAddedNodes() {
         assertThat(kafkaStatus.getAutoRebalance().getModes().get(0).getMode(), is(KafkaAutoRebalanceMode.ADD_BROKERS));
         assertThat(kafkaStatus.getAutoRebalance().getModes().get(0).getBrokers().stream().collect(Collectors.toSet()).equals(Set.of(3, 4, 5, 6)), is(true));
     }
+
+    @Test
+    public void testUpdateStatusCondition() {
+        KafkaRebalanceStatus status = new KafkaRebalanceStatusBuilder().withConditions(new ArrayList<>()).build();
+
+        // Test Warning condition can be added when there are no existing conditions
+        KafkaRebalanceUtils.addWarningCondition(status, exception("Example error 0"));
+        assertThat(status.getConditions().size(), is(1));
+        assertThat(KafkaRebalanceUtils.getWarningCondition(status).getType(), is("Warning"));
+
+        // Test Warning condition addition doesn't remove existing conditions.
+        Condition c0 = new ConditionBuilder()
+                .withType("Rebalancing")
+                .build();
+        status.setConditions(List.of(c0));
+        KafkaRebalanceUtils.addWarningCondition(status, exception("Example error 0"));
+        assertThat(status.getConditions().size(), is(2));
+        assertThat(KafkaRebalanceUtils.getWarningCondition(status).getType(), is("Warning"));
+
+        // Test original Warning is kept when a new error with same reason and message is thrown.
+        String timestamp = "2024-11-05T15:28:23.995129903Z";
```

Comment:
If this is an example of the timestamp format that CC gives you then all the timestamps are in UTC. That is what the `Z` at the end means (zero offset or "Zulu" time if you are in the military). Given that is the case you should be using UTC throughout NOT whatever the system local timezone is.